lib/savolvw.rb

```diff
@@ -28,22 +29,23 @@ def parse
       parsed_email = JSON.parse(@email.body.gsub('!@#', '')) rescue nil
 
       if parsed_email.nil?
-        parsed_email = @email.body.colons_to_hash(/(Telefone|Nome|Mensagem|E-mail|CPF).*?:/, false) unless parsed_email
+        parsed_email = @email.body.colons_to_hash(/(Telefone|Nome|Mensagem|E-mail|CPF|Campanha|Origem|ATENÇÃO).*?:/, false)
+        source_name = parsed_email['origem'] ? parsed_email['origem'] : F1SalesCustom::Email::Source.all[1][:name]
 
         {
           source: {
-            name: F1SalesCustom::Email::Source.all[1][:name]
+            name: source_name
           },
           customer: {
             name: parsed_email['nome'],
-            phone: parsed_email['telefone'],
+            phone: parsed_email['telefone'].gsub(/[^0-9]/, ''),
```

Comment:
The phone comes with the country code area, but I keep it, because the `app/models/customer.rb` from `f1sales-api` already removes the country code before saving the customer in the database.